back/app/models/basket.rb

```diff
@@ -28,10 +28,14 @@ class Basket < ApplicationRecord
   has_many :baskets_ideas, -> { order(:created_at) }, dependent: :destroy, inverse_of: :basket
   has_many :ideas, through: :baskets_ideas
 
+  # TODO: Others use nullify and a before destroy function, but can't work out why
```

Comment:
Have we any idea why others use nullify and a before destroy function?

Reply:
It's been a long time ago, but I think it's because there are two scenarios:
1. There is a presence validation on the notification (for example the comment for a comment on your idea notification). This is because the notification is not meaningful without this relationship. When the object (comment) is deleted, the notification should therefore also be deleted. This is why we try an update to set the relationship to null and delete the notification if that update fails (because the presence validation triggered an error).
2. The relationship is optional for the notification (for example the initiating user). If the object (initiating user) is deleted, the notification needs to be retained.

If at some point, we want to keep notifications that referenced to a deleted basket, some code like this would be necessary. The basket model is also not the right place to make this assumption, which is why my preference would be to add the nullify and before destroy to deal with that.

In general, notifications and emails deserve some refactoring so we don't need to write this code everywhere.